<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>973</width>
    <height>490</height>
   </rect>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>Compression</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QRadioButton" name="radioHuffman">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>20</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>Huffman Coding</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioArith">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>40</y>
      <width>141</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>Arithmetic Coding</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelAlphabet">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>60</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Alphabet</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelInput">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>140</y>
      <width>151</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Input</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textOutput">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>591</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Monaco</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>ForbiddenCursor</cursorShape>
    </property>
   </widget>
   <widget class="QLabel" name="labelOutput">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>230</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Encoded</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="textAlphabet">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Monaco</family>
      <pointsize>12</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="btnGen">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>100</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="textInput">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>591</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Monaco</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="documentTitle">
     <string/>
    </property>
    <property name="plainText">
     <string/>
    </property>
    <property name="backgroundVisible">
     <bool>false</bool>
    </property>
    <property name="placeholderText">
     <string/>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_1">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>70</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>70</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>70</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>70</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="btnAdjust">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>100</y>
      <width>81</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Adjust</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioBlock">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>40</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>Block Coding</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textDecoded">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>591</width>
      <height>65</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Monaco</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>ForbiddenCursor</cursorShape>
    </property>
   </widget>
   <widget class="QLabel" name="labelDecoded">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>330</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Decoded</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioNoCoding">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>No Coding</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_8">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>100</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>100</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_7">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>100</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>100</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_EOT">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>130</y>
      <width>62</width>
      <height>24</height>
     </rect>
    </property>
    <property name="decimals">
     <number>3</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QCheckBox" name="adjustCheckbox">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>70</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Automatic</string>
    </property>
   </widget>
   <widget class="QCalendarWidget" name="calendarWidget">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>20</y>
      <width>312</width>
      <height>173</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="optPerfLabel">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>300</y>
      <width>130</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Target Performance:</string>
    </property>
   </widget>
   <widget class="QLabel" name="actualPerfLabel">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>330</y>
      <width>130</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string> Actual Performance:</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="optPerfText">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>300</y>
      <width>110</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>ForbiddenCursor</cursorShape>
    </property>
   </widget>
   <widget class="QTextBrowser" name="actPerfText">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>330</y>
      <width>110</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>ForbiddenCursor</cursorShape>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioShannon">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>20</y>
      <width>140</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>Shannon Coding</string>
    </property>
   </widget>
   <widget class="QLabel" name="avglenLabel">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>240</y>
      <width>110</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string> Average Length:</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="avglenText">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>240</y>
      <width>110</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>ForbiddenCursor</cursorShape>
    </property>
   </widget>
   <widget class="QTextBrowser" name="entropyText">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>270</y>
      <width>110</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>ForbiddenCursor</cursorShape>
    </property>
   </widget>
   <widget class="QLabel" name="entropyLabel">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>270</y>
      <width>60</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string> Entropy:</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioLZ77">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>20</y>
      <width>90</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>LZ77</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioLZW">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>40</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="text">
     <string>LZW</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>973</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
